### PR TITLE
Add tests for the Dataflow Mark Duplicates by running the existing Mark Duplicates tests on the Dataflow code.

### PR DESCRIPTION
**Pull request by tovanadler**

### CODE REVIEW

File: src/test/java/org/broadinstitute/hellbender/utils/read/testers/SamFileTester.java [hunk 1]
```diff
@@ -255,18 +262,22 @@ public void addMatePair(final String readName,
      */
     public void runTest() {
         final File input = createInputFile();
-
         output = new File(outputDir, "output.sam");
-        addArg("--INPUT", input.getAbsolutePath());
-        addArg("--OUTPUT", output.getAbsolutePath());
+        if (useStandardInputNames) {
```
Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/utils/read/testers/SamFileTester.java]: I don't like that we have to have this switch here.  Could you open a ticket to update tools so that it is unnecessary? Subtask of #133

File: src/test/java/org/broadinstitute/hellbender/utils/read/markduplicates/AbstractMarkDuplicatesCommandLineProgramTest.java [hunk 2]
```diff
@@ -20,16 +20,22 @@
 
     protected abstract AbstractMarkDuplicatesTester getTester();
 
+  // ELIGIBLE_QUALITY is the minimum quality considered by the dataflow version
+  // for marking a specific pair as best. We use it to ensure that the expected
+  // pair wins the tie so that the tie is not broken randomly.
     protected final static int DEFAULT_BASE_QUALITY = 10;
+    protected final static int ELIGIBLE_QUALITY = 15;
```
Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/utils/read/markduplicates/AbstractMarkDuplicatesCommandLineProgramTest.java]: Maybe "BETTER_BASE_QUALITY" would be clearer?

File: src/test/java/org/broadinstitute/hellbender/utils/read/markduplicates/AbstractMarkDuplicatesCommandLineProgramTest.java [hunk 3]
```diff
@@ -20,16 +20,22 @@
...
 
-    @Test
+  /** Test failure: No records output */
+    @Test(enabled = false)
```
Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/utils/read/markduplicates/AbstractMarkDuplicatesCommandLineProgramTest.java]: Instead of disabling these in the superclass, can we override them in the dataflow subclass to make them no-ops?  That way it doesn't hurt the existing mark duplicates suite until we fix them

File: src/test/java/org/broadinstitute/hellbender/utils/read/testers/SamFileTester.java (hunk 1 above)
Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/utils/read/testers/SamFileTester.java]: I just made it.  #805.

**Comment by lbergelson** [comment]
@tovanadler Review complete.  Lets only disable the tests for dataflow.  Thanks for including the test failure reason in there.

File: src/test/java/org/broadinstitute/hellbender/utils/read/testers/SamFileTester.java (hunk 1 above)
Comment (tovanadler) [inline src/test/java/org/broadinstitute/hellbender/utils/read/testers/SamFileTester.java]: Thanks! Seemed like the least invasive way for now, but it's definitely a hack.

File: src/test/java/org/broadinstitute/hellbender/utils/read/markduplicates/AbstractMarkDuplicatesCommandLineProgramTest.java (hunk 2 above)
Comment (tovanadler) [inline src/test/java/org/broadinstitute/hellbender/utils/read/markduplicates/AbstractMarkDuplicatesCommandLineProgramTest.java]: Sure. This is what is set as MIN_QUAL in the code, and it's the minimum quality that is even considered for scoring. Happy to switch it, if you think BETTER_BASE_QUALITY is still clearer.

Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/utils/read/markduplicates/AbstractMarkDuplicatesCommandLineProgramTest.java]: Oh, I didn't understand, but re-reading the note that makes sense.  I was thinking it just had to be higher to ensure that it was the best, not that it was meeting some threshold.  Sorry, reading failure on my part.

Comment (tovanadler) [inline src/test/java/org/broadinstitute/hellbender/utils/read/markduplicates/AbstractMarkDuplicatesCommandLineProgramTest.java]: I changed it to ELIGIBLE_BASE_QUALITY. Think that is the clearest.

**Comment by tovanadler** [comment]
Thanks for the review. Back at you Louis.

**Comment by lbergelson** [comment]
:+1: Looks good.  

Squash these down to a single commit and you're ready to merge.

**Comment by akiezun** [comment]
@jean-philippe-martin can you take over? If this done? (it seems to so)

**Comment by jean-philippe-martin** [comment]
@akiezun I think you're asking me to squash/merge this code. Is this right?

**Comment by akiezun** [comment]
yes, thanks. rebase+squash and merge

**Comment by jean-philippe-martin** [comment]
rebase+squash. Merge pending green tests.
